File: Treinamento Calouros/2024/Turma 2/Contests/Simulado Primeira Fase OBI/pares.py
#https://neps.academy/br/exercise/8 - Pares de Números

n, I, F = [int(i) for i in input().split()]
v = [int(i) for i in input().split()]

resp = 0

for i in range(n):
    for j in range(i+1, n):
        if v[i]+v[j]>=I and v[i]+v[j]<=F:
            resp+=1

print(resp)

File: Treinamento Calouros/2024/Turma 2/Contests/Simulado Primeira Fase OBI/pares.cpp
//https://neps.academy/br/exercise/565 - Pares de Números

#include <bits/stdc++.h>

using namespace std;

#define endl '\n'

int main(){
    ios_base::sync_with_stdio(false), cin.tie(nullptr);
    int n, I, F, v[1005];
    cin >> n >> I >> F;

    for(int i=0; i<n; i++){
        cin >> v[i];
    }
    int resp = 0;
    for(int i=0; i<n; i++){
        for(int j=i+1; j<n; j++){
            if(v[i]+v[j]>=I && v[i]+v[j]<=F){
                resp++;
            }
        }
    }
    cout << resp << endl;
    return 0;
}